loki/transformations/tests/test_data_offload.py

```diff
@@ -805,3 +859,457 @@ def test_transformation_global_var_derived_type_hoist(here, config, frontend, ho
+            'in another_kernel') in caplog.records[0].message
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_field_offload_warnings(caplog, frontend, parkind_mod, field_module, tmp_path):
+    fother_state = """
+    module state_type_mod
+      implicit none
+      type state_type2
+        real, dimension(10,10), pointer :: a, b, c
+      contains
+        procedure :: update_view => state_update_view
+      end type state_type2
+
+    contains
+
+      subroutine state_update_view(self, idx)
+        class(state_type2), intent(in) :: self
+        integer, intent(in)           :: idx
+      end subroutine
+    end module
+    """
+    fother_mod= """
+    module another_module
+      implicit none
+    contains
+      subroutine another_kernel(nlon, nlev, a, b, c)
+        integer, intent(in)             :: nlon, nlev
+        real, intent(in)     :: a(nlon,nlev)
+        real, intent(inout)  :: b(nlon,nlev)
+        real, intent(out)    :: c(nlon,nlev)
+        integer :: i, j
+      end subroutine
+    end module
+    """
+    fcode = """
+    module driver_mod
+      use state_type_mod, only: state_type2
+      use parkind1, only: jprb
+      use field_module, only: field_2rb, field_3rb
+      use another_module, only: another_kernel
+
+      implicit none
+
+      type state_type
+        real(kind=jprb), dimension(10,10), pointer :: a, b, c
+        class(field_3rb), pointer :: f_a, f_b, f_c
+        contains
+        procedure :: update_view => state_update_view
+      end type state_type
+
+    contains
+
+      subroutine state_update_view(self, idx)
+        class(state_type), intent(in) :: self
+        integer, intent(in)           :: idx
+      end subroutine
+
+      subroutine kernel_routine(nlon, nlev, a, b, c)
+        integer, intent(in)             :: nlon, nlev
+        real(kind=jprb), intent(in)     :: a(nlon,nlev)
+        real(kind=jprb), intent(inout)  :: b(nlon,nlev)
+        real(kind=jprb), intent(out)    :: c(nlon,nlev)
+        integer :: i, j
+
+        do j=1, nlon
+          do i=1, nlev
+            b(i,j) = a(i,j) + 0.1
+            c(i,j) = 0.1
+          end do
+        end do
+      end subroutine kernel_routine
+
+      subroutine driver_routine(nlon, nlev, state, state2)
+        integer, intent(in)             :: nlon, nlev
+        type(state_type), intent(inout) :: state
+        type(state_type2), intent(inout) :: state2
+
+        integer                         :: i
+        real(kind=jprb)                 :: a(nlon,nlev)
+        real, pointer                   :: loki_devptr_prefix_state_b
+
+        !$loki data
+        do i=1,nlev
+            call state%update_view(i)
+            call kernel_routine(nlon, nlev, a, state%b, state2%c)
+        end do
+        !$loki end data
+
+      end subroutine driver_routine
+    end module driver_mod
+    """
+    with caplog.at_level(log_levels['WARNING']):
```

Comment:
Again, ideally keep the caplog tightly around the call you're expecting to trigger this. Otherwise unrelated constructor changes can make this test fail very easily.

Reply:
Ahhh, ignore this! I had misunderstood the caplog fixture and the log-level management in `.at_level`. Please ignore and leave this as is.